fetch/mesh_data.go

```diff
@@ -449,3 +439,30 @@ func (b *BatchError) IsIgnored(hash types.Hash32) bool {
 	}
 	return errors.Is(err, pubsub.ErrValidationReject) || errors.Is(err, ErrIgnore)
 }
+
+func readIDSlice[V any, H scale.DecodablePtr[V]](r io.Reader, slice *[]V, limit uint32) (int, error) {
```

Comment:
I feel like the generic option here isn't really needed because:

We can directly check `respLen` in the callback of `server.ReadResponse` if it is bigger than expected and abort immediately and then limit the decoding process to `respLen` bytes using a `io.LimitReader` which returns an `io.ErrUnexpectedEOF` if the decoder tries to read beyond `respLen` bytes.

Reply:
As discussed: `readIDSlice` also verifies consistency between outer `Response` length and the slice length. Also, there are complications related to variable-length encoding being used for the slice length